wordpress-mobile/WordPress-iOS
# Remove CocoaLumberjack from the libraries

## Changes

### Library updates

Here are full change sets in the libraries:
- WordPressShared: [1.18.0 -> 2.0.0-beta.2](https://github.com/wordpress-mobile/WordPress-iOS-Shared/compare/1.18.0...2.0.0-beta.2)
- WordPressKit: [4.58.2 -> 5.0.0](https://github.com/wordpress-mobile/WordPressKit-iOS/compare/4.58.2...5.0.0)
- WordPressAuthenticator: [3.2.2 -> 5.0.0](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/compare/3.2.2...5.0.0)

Unfortunately the CHANGELOG file doesn't contain all the releases. But by looking at the diff, WordPressShared and WordPressKit updates seems simply and straight forward to me. The WordPressAuthenticator update brings in some feature changes, which seems are all additive changes according to its [changelog](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/CHANGELOG.md).

### Logging in the libraries

This is the main reason I created this PR.

WordPressShared, WordPressKit, and WordPressAuthenticator used to depend on CocoaLumberjack and have their own `ddLogLevel` variable. It's strange that compiler would allow this, but this approach feels problematic anyway.

The new version of these libraries now have a function to set a logger instance, which uses the CocoaLumberjack set up in the app. Thus they no longer depend on CocoaLumberjack and also makes changing log level easier—there is only one log level variable which is in the app.

I've also updated the CocoaLumberjack Swift logging functions implementation in the app, using built-in functions in the latest version rather than using a copy of these functions in our own codebase.

## Test instructions

Not entirely what to test. Maybe make sure log in and log out work as expected (which I've checked)?

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
